dev_tools/oneapi/convert_source.sh

```diff
@@ -0,0 +1,429 @@
+#!/bin/bash
+
+# the following parameters set by environment
+# MUST:
+#   CUDA_HEADER_DIR: contains the cuda headers
+# OPTIONAL:
+#   ROOT_DIR: the ginkgo folder. The default is current path
+#   BUILD_DIR: the ginkgo build folder, which needs cmake to generate config.hpp and gtest include. The default is "build"
+#              Note. It requires GINKGO_BUILD_TESTS=ON to download gtest but does not need to compile ginkgo.
+#                    If GTEST_HEADER_DIR is from other place, does not require GINKGO_BUILD_TESTS.
+#                    If copy the ginkgo config.hpp from other ginkgo build into "${ROOT_DIR}/include/ginkgo/", do not require cmake.
+#   ROOT_BUILD_DIR: the complete path for build folder. The default is "${ROOT_DIR}/${BUILD_DIR}"
+#   GTEST_HEADER_DIR: the gtest header folder. The default is "${ROOT_BUILD_DIR}/third_party_gtest/src/googletest/include"
+#   CLANG_FORMAT: the clang-format exec. The default is "clang-format"
+CURRENT_DIR="$( pwd )"
+cd "$( dirname "${BASH_SOURCE[0]}" )"
+SCRIPT_DIR="$( pwd )"
+ROOT_DIR="${ROOT_DIR:="${CURRENT_DIR}"}"
+BUILD_DIR="${BUILD_DIR:="build"}"
+ROOT_BUILD_DIR="${ROOT_BUILD_DIR:="${ROOT_DIR}/${BUILD_DIR}"}"
+CUDA_HEADER_DIR="${CUDA_HEADER_DIR}"
+GTEST_HEADER_DIR="${GTEST_HEADER_DIR:="${ROOT_BUILD_DIR}/third_party/gtest/src/googletest/include"}"
+CLANG_FORMAT=${CLANG_FORMAT:="clang-format"}
+if [[ "${VERBOSE}" == 1 ]]; then
+    echo "the current setting is "
+    echo "CURRENT_DIR ${CURRENT_DIR}"
+    echo "SCRIPT_DIR ${SCRIPT_DIR}"
+    echo "ROOT_DIR ${ROOT_DIR}"
+    echo "ROOT_BUILD_DIR ${ROOT_BUILD_DIR}"
+    echo "GTEST_HEADER_DIR ${GTEST_HEADER_DIR}"
+    echo "CUDA_HEADER_DIR ${CUDA_HEADER_DIR}"
+    echo "ROOT_BUILD_DIR ${ROOT_BUILD_DIR}"
+    echo "CLANG_FORMAT ${CLANG_FORMAT}"
+fi
+if [[ "${CUDA_HEADER_DIR}" == "" ]]; then
+    echo "Please set the CUDA_HEADER_DIR in environment"
+    exit 1
+fi
+# move to working_directory
+cd working_directory
```

Comment:
Where is this supposed to happen, in the source directory?
Also, there is no `mkdir` for this, shouldn't there be?

Reply:
the working directory is put in the dev_tools/oneapi
It is happened in the script dir. we change directory to script dir in the begining.